iam_groups_authn/sync.py

```diff
@@ -130,6 +133,49 @@ async def insert_db_user(self, user_email, instance_connection_name):
                 f"Error: Failed to add IAM user `{user_email}` to Cloud SQL database instance `{instance_connection_name.instance}`."
             ) from e
 
+    async def verify_db_version(self, instance_connection_name):
+        """Verify database version of a Cloud SQL instance is supported.
+
+        Args:
+            instance_connection_name: InstanceConnectionName namedTuple.
+                (e.g. InstanceConnectionName(project='my-project', region='my-region',
+                instance='my-instance'))
+
+        Returns:
+            database_version: Database version of given Cloud SQL instance.
+        """
+        # build request to SQL Admin API
+        project = instance_connection_name.project
+        region = instance_connection_name.region
+        instance = instance_connection_name.instance
+        url = f"https://sqladmin.googleapis.com/sql/v1beta4/projects/{project}/instances/{instance}"
+
+        try:
+            # call the SQL Admin API
+            resp = await authenticated_request(
+                self.creds, url, self.client_session, RequestType.get
+            )
+            results = json.loads(await resp.text())
+
+            # check if database version is supported
+            database_version = results.get("databaseVersion", None)
+            if database_version not in SUPPORTED_DATABASES:
+                raise ValueError
+            logging.debug(
+                "[%s:%s:%s] Database version found: %s"
+                % (project, region, instance, database_version)
+            )
+            return database_version
+
+        except ValueError as e:
+            raise ValueError(
+                f"Unsupported database version for instance `{instance_connection_name}`. Current supported versions are: {SUPPORTED_DATABASES}"
+            ) from e
+        except Exception as e:
+            raise Exception(
+                f"Error: Failed to get the database version for `{instance_connection_name}`. Verify instance connection name and instance details."
+            ) from e
```

Comment:
This is a bit of a hacky pattern, but you can clean it up by moving the actually check out of the try/catch section


So I'd suggest either changing this function to `get_database_version` and move the check into the main flow, or else just move the if statement out of try/catch and don't worry about returning